upgrade terraform provider google beta to v6.23.0

t0yv0 opened the pull request:
Upgrade terraform provider google beta to v6.23.0 

There were compilation issues due to the provider's reliance on newly introduced ephemeral resources.

Since upstream is adding write_only attributes that pulumi-terraform-bridge cannot yet support, a patch is added to undo this addition. This also allows the project to keep the SDKv2 patch in sync with the version tested in the bridge.

Affected write_only attributes on resources:

- added password_wo and password_wo_version fields to google_sql_user resource (#21616)
- added secret_access_key_wo and secret_access_key_wo_version fields to google_bigquery_data_transfer_config resource (#21617)
- added secret_data_wo and secret_data_wo_version fields to google_secret_version resource (#21618)

New resources:

gemini/dataSharingWithGoogleSettingBinding.DataSharingWithGoogleSettingBinding
spanner/instancePartition.InstancePartition

New functions:

alloydb/getCluster.getCluster
projects/getAncestry.getAncestry

guineveresaenger commented on sdk/dotnet/Sql/User.cs:
```diff
@@ -151,6 +151,16 @@ namespace Pulumi.Gcp.Sql
+    /// 
+    /// The following write-only attributes are supported:
+    /// 
+    /// * `password_wo` - (Optional) The password for the user. Can be updated. For Postgres
```
We cannot support write-only attributes fully yet. I think the safest thing here to do would be to patch them until we can either suppress them more effectively (e.g. via the bridge) or until https://github.com/pulumi/pulumi/issues/16600 is implemented fully.

Patching is unfortunately the messiest option, but it's what we did for AWS: https://github.com/pulumi/pulumi-aws/issues/5231

t0yv0 commented:
Per @guineveresaenger https://github.com/pulumi/pulumi-aws/issues/5215 we cannot automatically handle wo attributes reliably therefore we might need to patch out these:

https://github.com/hashicorp/terraform-provider-google-beta/releases/tag/v6.23.0

```
SQL: added password_wo and password_wo_version fields to google_sql_user resource (#21616)
bigquerydatatransfer: added secret_access_key_wo and secret_access_key_wo_version fields to google_bigquery_data_transfer_config resource (#21617)
secretmanager: added secret_data_wo and secret_data_wo_version fields to google_secret_version resource (#21618)
```

t0yv0 commented:
Per @corymhall : prefer regular bridge update than updating the SDK forward. Trying this out.

guineveresaenger reviewed (changes requested):
We'll most likely want to add a patch to not include the read-only attributes right now. More details in comment on one such example.
I've relied on the [releasenotes](https://github.com/hashicorp/terraform-provider-google-beta/releases/tag/v6.23.0) to tell us which of these write-only attributes were added.

More context: https://github.com/pulumi/pulumi-aws/issues/5215

t0yv0 commented:
Since the bridge release is getting a tad delayed, I reworked this using patches.

VenelinMartinov commented:
@t0yv0 does this need an SDK regeneration?
